### ISSUE TITLE
Android 15 edge-to-edge support on 3DSecure screen

### ISSUE THREAD

**Issue by francescocervone** [braintree/braintree_android#1236]
### Braintree SDK Version

5.2.0

### Environment

Both

### Android Version & Device

Android 15

### Braintree dependencies

I tried the demo on the main's HEAD, which at the time of writing is [this commit](https://github.com/braintree/braintree_android/commit/e12b9858980852aa5bafa83d43be68b6dc86c8c3).

### Describe the bug

[Android 15 has forced edge to edge](https://developer.android.com/about/versions/15/behavior-changes-15#edge-to-edge) automatically for all activities.

The 3DSecure doesn't correctly support the Android 15 edge to edge enforcement.

We depend on Braintree 4.x and we were trying to increase our target SDK to 35 (Android 15), but we found out about this behavior. 

So, since the 5.x declares Android 15 support, we decided to upgrade to Braintree 5 first.

Unfortunately though, I tried the demo version on the main's HEAD, which at the time of writing is [this commit](https://github.com/braintree/braintree_android/commit/e12b9858980852aa5bafa83d43be68b6dc86c8c3), and that screen still draws incorrectly behind the system bars.

### To reproduce

- Run a 3DSecure verification on an app targeting Android API 35
- The 3DSecure screen is drawn behind the system bars incorrectly

### Expected behavior

- All Braintree screens handle window insets correctly in all Android versions, 15 included.

### Screenshots

![Screenshot_20241210_122321](https://github.com/user-attachments/assets/a0bbb28f-c704-429a-8853-a4288a76c4b2)

**Comment by tdchow** [comment]
Thanks again for reporting! We're tracking this internally.

**Comment by francescocervone** [comment]
Hi, just a message to help you understand the priority of this issue from our point of view.

Please keep in mind that apps have 1 year to be compliant with the [Target API level requirements for Google Play apps](https://support.google.com/googleplay/android-developer/answer/11926878)). 

Typically, that would be August 2025, but given that Google changed the frequency and dates of new SDK releases ([blog](https://android-developers.googleblog.com/2024/10/android-sdk-release-update.html)), apps will need to start integrating with Android 16 too starting from Q2 2025 (I guess around April 2025).

Thanks.

**Comment by tdchow** [comment]
Hi @francescocervone - Understood! We'll try to prioritize as soon as possible.

**Comment by francescocervone** [comment]
Hi @tdchow. Is there any news about this?
Thanks

**Comment by saperi22** [comment]
@francescocervone no, we don't have any updates on this. We are trying to work with Cardinal to see when this could be made available.
We'll keep you posted on any updates we have.

**Comment by saperi22** [comment]
@francescocervone Our Cardinal POC mentioned that they'd be releasing a 2.2.7-7 version that supports Android 15 by the end of this month. We'll let you know once we've received the update and are ready to prioritize the upgrade work.

**Comment by francescocervone** [comment]
Thanks @saperi22

**Comment by saperi22** [comment]
@francescocervone we've received an update from Cardinal that the 2.2.7-7 version is available. We'll be testing this dependency for issues in the next couple weeks and if everything looks good, we'll release it.

**Comment by saperi22** [comment]
https://github.com/braintree/braintree_android/pull/1282 is where the changes went it.

| Before    | After |
| -------- | ------- |
| <img width="426" alt="Image" src="https://github.com/user-attachments/assets/3dde6b6f-b4df-4474-a0ed-976c3b4ccf1b" />  |  <img width="427" alt="Image" src="https://github.com/user-attachments/assets/3377fe86-82ba-4ff6-b4a5-52e0a6ed3833" />  |

**Comment by jogwir** [comment]
Quick question about these changes as I've just tested the update in the latest Android drop-in library too and notice the same behaviour as your screenshot - the status bar content is not visible, you might need to apply a style to the content to ensure it is visible.

**Comment by noguier** [comment]
@jogwir Android drop-in is a deprecated library that is not supported anymore. I will close this issue as the initial fix has been released. Feel free to open another issue with more details if you need more support!
